packages/clay-form/src/MultiSelect.tsx

```diff
@@ -0,0 +1,283 @@
+
+const BACKSPACE_KEY = 8;
+const COMMA_KEY = 188;
+const ENTER_KEY = 13;
+
+const DELIMITER_KEYS = [ENTER_KEY, COMMA_KEY];
+
+export interface IProps extends React.HTMLAttributes<HTMLInputElement> {
+	/**
+	 * Message to display if isValid is false.
+	 */
+	errorMessage?: string;
+
+	/**
+	 * Ref added to container element
+	 */
+	forwardRef: React.RefObject<HTMLDivElement>;
+
+	/**
+	 * Message to display which helps users interact with input
+	 */
+	helpText?: string;
+
+	/**
+	 * Value of input
+	 */
+	inputValue: string;
+
+	/**
+	 * Values that display as label items
+	 */
+	items: string[];
+
+	/**
+	 * Description of the input
+	 */
+	label?: React.ReactText;
+
+	/**
+	 * Callback for when items are added or removed
+	 */
+	onItemsChange: (val: string[]) => void;
+
+	/**
+	 * Callback for when the input value changes
+	 */
+	onInputChange: (val: string) => void;
+
+	/**
+	 * List of pre-populated items that will show up in a dropdown menu
+	 */
+	sourceItems?: string[];
+
+	/**
+	 * Path to spritemap for clay icons
+	 */
+	spritemap?: string;
+
+	/**
+	 * Flag to indicate if input is valid or not
+	 */
+	isValid?: boolean;
+}
+
+const MultiSelect: React.FunctionComponent<IProps> = ({
+	errorMessage,
+	forwardRef,
+	helpText,
+	inputValue = '',
+	isValid = true,
+	items = [],
+	label,
+	onInputChange,
+	onItemsChange,
+	sourceItems,
+	spritemap,
+	...otherProps
+}) => {
+	const inputRef = useRef<HTMLInputElement>(null);
+	const lastItemRef = useRef<HTMLSpanElement>(null);
+	const [active, setActive] = useState(false);
+	const [isFocused, setIsFocused] = useState();
+
+	useLayoutEffect(() => {
+		if (sourceItems) {
+			const matchedItems = sourceItems.filter(
+				item => inputValue && item.match(inputValue)
+			);
+
+			setActive(matchedItems.length !== 0);
+		}
+	}, [sourceItems, inputValue]);
+
+	const setNewValue = (newVal: string) => {
+		onItemsChange([...items, newVal]);
+
+		onInputChange('');
+	};
+
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		const {keyCode} = event;
+
+		if (inputValue && DELIMITER_KEYS.includes(keyCode)) {
+			event.preventDefault();
+
+			setNewValue(inputValue);
+		} else if (
+			!inputValue &&
+			keyCode === BACKSPACE_KEY &&
+			inputRef.current &&
+			lastItemRef.current
+		) {
+			inputRef.current.blur();
+			lastItemRef.current.focus();
+		}
+	};
+
+	const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+		const pastedText = event.clipboardData.getData('Text');
+
+		const pastedItems = pastedText
+			.split(',')
+			.map(item => item.trim())
+			.filter(Boolean);
+
+		if (pastedItems.length > 0) {
+			event.preventDefault();
+
+			onItemsChange([...items, ...pastedItems]);
+		}
+	};
+
+	return (
+		<div className="form-group" ref={forwardRef}>
+			{label && <label>{label}</label>}
+
+			<div className="input-group input-group-stacked-sm-down">
+				<div
+					className={classNames('input-group-item', {
+						focus: isFocused && isValid,
```

Comment:
We should move the focus to `.form-control`, if it does, it will create a focus on the contour of the form as well.